src/terminal.rs

```diff
@@ -222,13 +222,47 @@ impl<H: HashT, NI: NodeIdT> Terminal<H, NI> {
     }
 
     fn update_on_dag_add(&mut self, u_hash: &H) {
-        let u = self.unit_store.get(u_hash).unwrap();
+        let u = self
+            .unit_store
+            .get(u_hash)
+            .expect("Unit to be added to dag must be in store");
         self.post_insert.iter().for_each(|f| f(u.clone()));
         if let Some(children) = self.children_hash.remove(u_hash) {
             for v_hash in children {
                 self.new_parent_in_dag(&v_hash);
             }
         }
+        let u = self
```

Comment:
We've just fetched `u` from `unit_store` above, so we may clone it if we need another copy.

Reply:
fixed, as discussed